drivers: pinmux: stm32: st: add L4 UART4 AF on PC10 and PC11

GeorgeCGV opened the pull request:
- Added UART4 alternate pin function for L4 µC for PC10 and PC11.
- Corrected naming of previously defined UART4 TX and RX defines.
  Was defined as U**S**ART but in reallity it is only UART.

Due to naming correction had to touch `pinmux.c` of `disco_l475_iot1` board.

GeorgeCGV commented:
@erwango

erwango commented on drivers/pinmux/stm32/pinmux_stm32l4x.h:
```diff
@@ -108,10 +108,14 @@
 #define STM32L4X_PINMUX_FUNC_PC10_USART3_TX \
 	(STM32_PINMUX_ALT_FUNC_7 | STM32_PUSHPULL_PULLUP)
+#define STM32L4X_PINMUX_FUNC_PC10_UART4_TX \
+	(STM32_PINMUX_ALT_FUNC_7 | STM32_PUSHPULL_PULLUP)
```
STM32_PINMUX_ALT_FUNC_8

erwango commented on drivers/pinmux/stm32/pinmux_stm32l4x.h:
```diff
@@ -108,10 +108,14 @@
 #define STM32L4X_PINMUX_FUNC_PC11_USART3_RX \
 	(STM32_PINMUX_ALT_FUNC_7 | STM32_PUPDR_NO_PULL)
+#define STM32L4X_PINMUX_FUNC_PC11_UART4_RX \
+	(STM32_PINMUX_ALT_FUNC_7 | STM32_PUPDR_NO_PULL)
```
STM32_PINMUX_ALT_FUNC_8

erwango commented on dts/arm/st/l4/stm32l4-pinctrl.dtsi:
```diff
@@ -70,6 +70,12 @@
+			uart4_pins_b: uart4_b {
+				rx_tx {
+					rx = <STM32_PIN_PC11 (STM32_PINMUX_ALT_FUNC_7 | STM32_PUPDR_NO_PULL)>;
+					tx = <STM32_PIN_PC10 (STM32_PINMUX_ALT_FUNC_7 | STM32_PUSHPULL_NOPULL)>;
```
STM32_PINMUX_ALT_FUNC_8 for both pins

Commit:
drivers: pinmux: stm32: st: add L4 UART4 AF on PC10 and PC11

Added UART4 alternate pin function for L4 µC for PC10 and PC11.
Corrected naming of previously defined UART4 TX and RX defines.

Signed-off-by: Georgij Cernysiov <g.cernysiov@elco-automation.de>

GeorgeCGV commented:
@erwango you are right it shall be AF8, corrected
